lib/brains/views/sale.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from lib.brains.client import get_client
+from lib.brains.errors import BraintreeResultError
+from lib.brains.forms import SaleForm
+from lib.brains.models import BraintreeTransaction
+from lib.brains.serializers import LocalTransaction, Namespaced
+from lib.transactions import constants
+from lib.transactions.models import Transaction
+from lib.transactions.serializers import TransactionSerializer
+from solitude.errors import FormError
+from solitude.logger import getLogger
+
+log = getLogger('s.brains')
+
+
+@api_view(['POST'])
+def create(request):
+    client = get_client().Transaction
+    form = SaleForm(request.DATA)
+
+    if not form.is_valid():
+        raise FormError(form.errors)
+
+    result = client.sale(form.braintree_data)
+    if not result.is_success:
+        log.warning('Error on one-off sale: {}'.format(form.braintree_data))
+        raise BraintreeResultError(result)
+
+    our_transaction = Transaction.objects.create(
+        amount=result.transaction.amount,
+        buyer=form.buyer,
+        currency=result.transaction.currency_iso_code,
+        provider=constants.PROVIDER_BRAINTREE,
+        seller=form.seller_product.seller,
+        seller_product=form.seller_product,
+        status=constants.STATUS_CHECKED,
+        type=constants.TYPE_PAYMENT,
+        uid_support=result.transaction.id
+    )
+    braintree_transaction = BraintreeTransaction.objects.create(
```

Comment:
how does the braintree transaction `kind` get set? Shouldn't we set that here since we asked to submit for settlement immediate?